src/base-api-client.ts

```diff
@@ -0,0 +1,90 @@
+import { ApiRequestHandler } from './api-request-handler';
+
+type ServiceApi = 'topic_extraction' | 'sentiment_analysis';
+
+/**
+ * Base client implementation. Intended to be extended by a specific client per API
+ */
+export class BaseApiClient<TJob, TResult> {
+    apiHandler: ApiRequestHandler;
+
+    /**
+     * @param accessToken Access token used to validate API requests
+     * @param serviceApi Type of api service
+     * @param version (optional) version of the API to be used
+     */
+    constructor (accessToken: string, serviceApi: ServiceApi, version = 'v1') {
+        this.apiHandler = new ApiRequestHandler(`https://api.rev.ai/${serviceApi}/${version}/`, accessToken);
+    }
+
+    /**
+     * Get information about a specific job
+     * @param id Id of job whose details are to be retrieved
+     * @returns Job details
+     */
+    protected async _getJobDetails(id: string): Promise<TJob> {
+        return await this.apiHandler.makeApiRequest<TJob>('get', `/jobs/${id}`, {}, 'json');
+    }
+
+    /**
+     * Get a list of jobs submitted within the last 30 days in reverse chronological order 
+     * (last submitted first) up to the provided limit number of jobs per call. Pagination is supported via passing
+     * the last job id from previous call into starting_after.
+     * @param params
+     * @returns List of job details
+     */
+    protected async _getListOfJobs(params?: {}): Promise<TJob[]> {
```

Comment:
These params are consistent across apis and should always be. I think we can have them hardcoded here

Reply:
I guess you did this for consistency with how the result call works. I prefer it the old way so we don't duplicate the logic of building this params dictionary in every implementation of this client

Reply:
Now that I left [this comment ](https://github.com/revdotcom/revai-node-sdk/pull/100/files#r878532135) I think it should look like this
```suggestion
    protected async _getListOfJobs(params?: GetListOfJobsOptions): Promise<TJob[]> {
```